File: 02-backend/spring-boot-ecommerce/src/main/java/com/example/ecommerce/dto/Purchase.java
package com.example.ecommerce.dto;

import com.example.ecommerce.entity.Address;
import com.example.ecommerce.entity.Customer;
import com.example.ecommerce.entity.Order;
import com.example.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    
    customer!: Customer
    shippingAddress!: Address;
    billingAddress!: Address;
    order!: Order;
    orderItems!: OrderItem[];

    // constructor(public customer: Customer,
    //     public shippingAddress: Address,
    //     public billingAddress: Address,
    //     public order: Order,
    //     public orderItems: OrderItem[]) {}
}
